Maple2.Server.Game/Manager/ItemBoxManager.cs

```diff
@@ -0,0 +1,284 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Maple2.Database.Storage;
+using Maple2.Model;
+using Maple2.Model.Enum;
+using Maple2.Model.Error;
+using Maple2.Model.Game;
+using Maple2.Model.Metadata;
+using Maple2.Server.Game.Session;
+
+namespace Maple2.Server.Game.Manager;
+
+public class ItemBoxManager {
+    #region Autofac Autowired
+    // ReSharper disable MemberCanBePrivate.Global
+    public required ItemMetadataStorage ItemMetadata { private get; init; }
+    public required TableMetadataStorage TableMetadata { private get; init; }
+    // ReSharper restore All
+    #endregion
+
+    private readonly GameSession session;
+    public int BoxCount;
+
+    public ItemBoxManager(GameSession session) {
+        this.session = session;
+        BoxCount = 0;
+    }
+
+    public void Reset() {
+        BoxCount = 0;
+    }
+
+    public ItemBoxError Open(Item item, int count = 1, int index = 0) {
+        int[] itemBoxParams = item.Metadata.Function?.Parameters.Split(',').Select(int.Parse).ToArray() ?? Array.Empty<int>();
+        return item.Metadata.Function?.Type switch {
+            ItemFunction.SelectItemBox => SelectItemBox(item, itemBoxParams, index, count),
+            ItemFunction.OpenItemBox => OpenItemBox(item, itemBoxParams, count),
+            ItemFunction.OpenItemBoxWithKey => OpenItemBoxWithKey(item, itemBoxParams, count),
+            _ => throw new ArgumentOutOfRangeException(item.Metadata.Function?.Type.ToString(), "Invalid box type"),
+        };
+    }
+
+    private ItemBoxError SelectItemBox(Item item, int[] itemBoxParams, int index, int count = 1) {
+        var error = ItemBoxError.ok;
```

Comment:
define variables where they will be used, in this case around L68

Reply:
what do you mean? some are intentionally returning in order to prevent the function from continuing

Reply:
the variable is never used until after L68 so define it later.